### PR TITLE
Monitor without JSON Conversion

### PR DESCRIPTION
**Pull request by vincentpierre**
* Fixed line indentation,
 * Removed the library Newtonsoft.Json from the monitor,
 * Replaced calls to JSON conversion with manual conversion

### CODE REVIEW

File: unity-environment/Assets/ML-Agents/Scripts/Monitor.cs [hunk 1]
```diff
@@ -194,34 +198,25 @@ static void InstantiateCanvas()
...
+        System.Type inputType = input.GetType();
+        System.Type expectedType = typeof(float);
+        if (inputType.IsArray &&
+            expectedType.IsAssignableFrom(inputType.GetElementType()))
         {
-
+            return input as float[];
         }
-
         return new float[0];
```
Comment (vladimiroster) [inline unity-environment/Assets/ML-Agents/Scripts/Monitor.cs]: Should that be legal? What's the use case of passing not a float here?

File: unity-environment/Assets/ML-Agents/Scripts/Monitor.cs [hunk 2]
```diff
@@ -194,34 +198,25 @@ static void InstantiateCanvas()
...
+        if (input.GetType() == typeof(List<float>))
         {
-            return new float[1]
-            {JsonConvert.DeserializeObject<float>(
-                    JsonConvert.SerializeObject(input, Formatting.None))
-            };
+            input = (input as List<float>).ToArray();
         }
-        catch
+        System.Type inputType = input.GetType();
```
Comment (vladimiroster) [inline unity-environment/Assets/ML-Agents/Scripts/Monitor.cs]: There must be a nicer way to do that. Why is the DisplayValue::value object? Can it be anything other than float/array of floats? Also unit testing this method will save you future grief.

Comment (vincentpierre) [reply unity-environment/Assets/ML-Agents/Scripts/Monitor.cs]: the value can be a string, a float or an array of float in the current implementation

Comment (vladimiroster) [reply unity-environment/Assets/ML-Agents/Scripts/Monitor.cs]: But if it's a string, you ignore it and return a float array, no?

File: unity-environment/Assets/ML-Agents/Scripts/Monitor.cs [hunk 3]
```diff
@@ -249,13 +244,19 @@ void OnGUI()
...
-            var origin = new Vector3(Screen.width / 2 - keyPixelWidth, Screen.height);
+            var origin = new Vector3(Screen.width / 2 -
+                                     keyPixelWidth, Screen.height);
             if (!(target == canvas.transform))
             {
-                Vector3 cam2obj = target.position - Camera.main.transform.position;
-                scale = Mathf.Min(1, 20f / (Vector3.Dot(cam2obj, Camera.main.transform.forward)));
-                Vector3 worldPosition = Camera.main.WorldToScreenPoint(target.position + new Vector3(0, verticalOffset, 0));
-                origin = new Vector3(worldPosition.x - keyPixelWidth * scale, Screen.height - worldPosition.y);
+                Vector3 cam2obj = target.position -
```
Comment (mmattar) [inline unity-environment/Assets/ML-Agents/Scripts/Monitor.cs]:
```
if (!(target == canvas.transform))
{
    Vector3 cam2obj = 
        target.position - Camera.main.transform.position;
    scale = Mathf.Min(
        1, 
        20f / (Vector3.Dot(cam2obj, Camera.main.transform.forward)));
    Vector3 worldPosition = Camera.main.WorldToScreenPoint(
        target.position + new Vector3(0, verticalOffset, 0));
    origin = new Vector3(
        worldPosition.x - keyPixelWidth * scale, Screen.height - worldPosition.y);
}
```

File: unity-environment/Assets/ML-Agents/Scripts/Monitor.cs [hunk 4]
```diff
@@ -42,8 +36,18 @@ public class Monitor : MonoBehaviour
...
+        public string stringValue;
+        public float floatValue;
+        public float[] floatArrayValues;
+        public enum ValueType
+        {
+            FLOAT,
+            FLOATARRAY_INDEPENDENT,
+            FLOATARRAY_PROPORTION,
+            STRING
+
```
Comment (awjuliani) [inline unity-environment/Assets/ML-Agents/Scripts/Monitor.cs]: Remove empty line.

**Pull request merged**